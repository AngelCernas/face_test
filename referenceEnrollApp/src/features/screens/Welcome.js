import React, { useEffect, useState } from 'react';

import {
  View,
  StyleSheet,
  Alert,
  Dimensions,
  Image,
  ImageBackground,
} from 'react-native';
import { Caption, Headline, Subheading1 } from '../../styles/fontStyles';
import CustomButton from '../../styles/CustomButton';
import { CONFIG } from '../../env/env.json';
import { useDispatch } from 'react-redux';
import * as constants from '../../shared/constants';
import { deletePersonGroup, validatePersonGroup } from '../../shared/helper';
var RNFS = require('react-native-fs');

function Welcome({ navigation }) {
  let dispatch = useDispatch();

  let responsiveStyles = {
    whiteBoxHeight: "60%",
    imgContainerHeight: 375,
    whiteBoxFlex: 1,
    rightBoxFlex: 1,
    leftBoxFlex: 1,
    buttonDirection: 'column',
    buttonContext: 'center'
  };

  const checkIsPortrait = () => {
    const dim = Dimensions.get('window');
    return dim.height >= dim.width;
  };

  const determineScreenWidth = () => {
    const dim = Dimensions.get('window');
    if (dim.width <= 600) {
      responsiveStyles = {
        whiteBoxHeight: "60%",
        imgContainerHeight: dim.height / 2,
        whiteBoxFlex: 0,
        rightBoxFlex: 1,
        leftBoxFlex: 1,
        buttonDirection: 'column',
        buttonContext: 'center',
        buttonFlex: 0,
      };
      return 'small';
    }
    else if (dim.width > 600 && dim.width < 840) {
      console.log('medium');

      responsiveStyles = {
        whiteBoxHeight: "30%",
        imgContainerHeight: dim.height,
        whiteBoxFlex: 10,
        rightBoxFlex: 1,
        leftBoxFlex: 1,
        buttonDirection: 'row',
        buttonContext: 'flex-start',
        buttonFlex: 1,
      };

      return 'medium';
    }
    else {
      console.log('large');

      responsiveStyles = {
        whiteBoxHeight: "70%", // this needs chanding
        imgContainerHeight: dim.height,
        whiteBoxFlex: 3, // this needs changing 
        rightBoxFlex: 7,
        leftBoxFlex: 2,
        buttonDirection: 'column',
        buttonContext: 'center',
        buttonFlex: 0,
      };
      return 'large';
    }

  }

  const [isPortrait, setIsPortrait] = useState(checkIsPortrait());
  const [screenWidth, setScreenWidth] = useState(determineScreenWidth());

  const showAlert = () => {
    // For development and testing environment, expose settings page
    let buttonOption =
      CONFIG.ENVIRONMENT == 'dev'
        ? [
          {
            text: 'Settings',
            onPress: () => navigation.navigate('Settings'),
          },
        ]
        : [
          {
            text: 'Try again',
            onPress: async () => {
              let validated = await validatePersonGroup(
                CONFIG.PERSONGROUP_RGB,
              );

              if (validated == false) {
                showAlert();
              }
            },
          },
        ];
    Alert.alert(
      'A problem occurred',
      'Cannot connect to service',
      buttonOption,
      {
        cancelable: false,
      },
    );
  };

  useEffect(() => {
    const orientationCallback = () => {
      setIsPortrait(checkIsPortrait());
      setScreenWidth(determineScreenWidth());
    };
    Dimensions.addEventListener('change', orientationCallback);

    validatePersonGroup(CONFIG.PERSONGROUP_RGB).then((personGroupValidated) => {
      if (personGroupValidated == false) {
        showAlert();
      }
    });

    return () => {
      Dimensions.removeEventListener('change', orientationCallback);
    };
  }, []);

  // This is for testing purposes
  const clearAllData = () => {
    RNFS.readDir(RNFS.DocumentDirectoryPath + '/enrollment/')
      .then((result) => {
        console.log('files', result);

        // stat the first file
        return result;
      })
      .then((result) => {
        for (let item of result) {
          if (item.isFile()) {
            console.log('deleting', item.path);
            RNFS.unlink(item.path);
          }
        }
      })
      .catch((err) => {
        console.log(err.message, err.code);
      });

    deletePersonGroup(CONFIG.PERSONGROUP_RGB).then((res) => {
      console.log('Delete result:', res);
    });
  };

  return (
    <View style={styles.container}>
      <View style={styles.backgroundColumn}>

        <View style={styles.backroundTopRow}>
          <View style={[styles.imgContainer, { height: responsiveStyles.imgContainerHeight, width: responsiveStyles.imgContainerHeight * 1.6 }]}>
            <Image
              source={require('../../assets/bg_heroIllustration_welcome.png')}
              style={styles.backgroundImage}
            />
          </View>
        </View>

        {isPortrait ? <View style={styles.backgroundBottomRow} /> : <View />}



        <View style={isPortrait ? styles.boxContainerP : styles.boxContainerL}>
          <View style={styles.leftBox, { flex: responsiveStyles.leftBoxFlex }} />
          <View
            style={[styles.whiteBox, { height: responsiveStyles.whiteBoxHeight, flex: responsiveStyles.whiteBoxFlex }]}>


            <View style={styles.infoView}>
              <View style={styles.textPadding}>
                <Headline>An easier way to get into work</Headline>
              </View>
              <Subheading1 style={styles.greyText}>
                You can now use face recognition instead of your badge to
                conveniently unlock building doors.
            </Subheading1>
            </View>

            {/* changes column to row, center to stert */}
            <View style={{ flexDirection: responsiveStyles.buttonDirection, flex: 2, justifyContent: responsiveStyles.buttonContext }}>

              <View style={[styles.buttons, { flex: responsiveStyles.buttonFlex }]}>
                <CustomButton
                  title="Get started"
                  onPress={() => {
                    dispatch({ type: 'USER_LOGOUT' });
                    navigation.navigate(constants.SCREENS.consent);
                  }}
                />
              </View>
              <View style={[styles.buttons, { flex: responsiveStyles.buttonFlex }]}>
                <CustomButton
                  whiteButton="true"
                  title="Manage profile"
                  onPress={() => {
                    dispatch({ type: 'USER_LOGOUT' });
                    navigation.navigate(constants.SCREENS.login, {
                      nextScreen: constants.SCREENS.manage,
                    });
                  }}
                />
              </View>
            </View>


            {/* This is for testing purposes */}
            {/* <View style={styles.buttons}>
            <CustomButton
              whiteButton="true"
              title="Delete all data"
              onPress={clearAllData}
            />
          </View> */}

            <View style={{ flex: 1, justifyContent: 'flex-end' }}>
              <Caption style={styles.greyText}>
                Details at contoso.com/touchless-access{'\n'}
                  Contoso Privacy Statement {'\n'}
              </Caption>

              <Caption style={styles.greyText}>
                Contoso | Real Estate & Security
              </Caption>
            </View>



          </View>

          <View style={styles.rightBox, { flex: responsiveStyles.rightBoxFlex }} />
        </View>


      </View>


    </View >
  );
}

var styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: '#002C55'
  },
  backgroundColumn: {
    flex: 1,
    flexDirection: 'column',
  },
  backroundTopRow: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: "flex-end",
  },
  imgContainer: {
    marginTop: -30,
    height: 375, // Image has a 1.6 aspect ratio
    width: 600,  // 375 X 600 = 1.6 AR
  },
  backgroundImage: {
    flex: 1,
    width: '100%',
    resizeMode: "contain",
  },
  backgroundBottomRow: {
    flex: 1,
  },
  boxContainerP: {
    flex: 12,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'flex-end',
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
  },
  whiteBox: {
    flex: 0, // change 0, 3 to 10
    flexDirection: 'column',
    backgroundColor: 'white',
    margin: 16,
    padding: 24,
    borderRadius: 4,
  },
  rightBox: {
    flex: 7, // changes 1 or 7
  },
  leftBox: {
    flex: 2, // changes, 1 or 2
  },
  buttons: {
    flex: 0, // doesn't allow wrapping 0 or 1
    paddingRight: 20,
    marginBottom: 20,
  },
  infoView: {
    flex: 2,
    justifyContent: "flex-end",
    marginBottom: 24,
  },
  textPadding: {
    justifyContent: "flex-end",
    paddingBottom: 16
  },
  boxContainerL: {
    flex: 12,
    flexDirection: 'row',
    position: 'absolute',
    alignItems: "center",
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
  },
  greyText: {
    color: '#77787B',
  },
});

export default Welcome;
