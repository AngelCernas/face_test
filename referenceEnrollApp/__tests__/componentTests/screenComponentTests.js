import 'react-native';
import React from 'react';
import {create, act} from 'react-test-renderer';
import Welcome from '../../src/features/screens/Welcome';
import Consent from '../../src/features/screens/Consent';
import Instruction from '../../src/features/screens/Instruction';
import ImageCapture from '../../src/features/screens/ImageCapture';
import Receipt from '../../src/features/screens/Receipt';
import ManageProfile from '../../src/features/screens/ManageProfile';
import {SCREENS} from '../../src/shared/constants';
import {render, fireEvent} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Login from '../../src/features/screens/Login';
import '@testing-library/jest-native/extend-expect';
import {StackActions} from '@react-navigation/native';

// Mocked store for screen component tests
const mockStore = configureMockStore();
const store = mockStore({
  userInfo: {rgbPersonId: 123},
  newEnrollment: {},
  feedback: {message: ''},
  newEnrollment: {},
});

// Mocked functions for testing navigation
const navigate = jest.fn();
const setOptions = jest.fn();
const dispatch = jest.fn();

// Mock fetch so person group check succeeds
global.fetch = jest.fn().mockImplementation(() =>
  Promise.resolve({
    status: 200,
    text: function () {
      return JSON.stringify('');
    },
  }),
);

describe('Welcome screen', () => {
  it('renders correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Welcome navigation={{navigate}} />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('navigates to consent page', () => {
    const {getByText} = render(
      <Provider store={store}>
        <Welcome navigation={{navigate}} />
      </Provider>,
    );
    fireEvent.press(getByText('Get started'));
    expect(navigate).toHaveBeenCalledWith(SCREENS.consent);
  });

  it('navigates to login page', () => {
    const {getByText} = render(
      <Provider store={store}>
        <Welcome navigation={{navigate}} />
      </Provider>,
    );
    fireEvent.press(getByText('Manage profile'));
    expect(navigate).toHaveBeenCalledWith(SCREENS.login, {
      nextScreen: SCREENS.manage,
    });
  });
});

describe('Consent screen', () => {
  it('navigates to login', () => {
    const {getByText} = render(
      <Consent navigation={{navigate, setOptions}}></Consent>,
    );
    fireEvent.press(getByText('Yes, create my face template'));
    expect(navigate).toHaveBeenCalledWith(SCREENS.login, {
      nextScreen: SCREENS.instruction,
    });
  });

  it('navigates back to welcome', async () => {
    const {getByText, findByText, queryByText} = render(
      <Consent navigation={{dispatch, setOptions}}></Consent>,
    );

    let modal = queryByText('Got it');
    expect(modal).toBeNull();

    fireEvent.press(getByText('No, donâ€™t create my face template'));

    modal = await findByText('Got it');
    expect(modal).toBeTruthy();

    fireEvent.press(getByText('Close'));
    expect(dispatch).toHaveBeenCalledWith(StackActions.popToTop());
  });
});

describe('Login screen', () => {
  it('requires username input', async () => {
    const {getByText, queryByText} = render(
      <Provider store={store}>
        <Login
          navigation={{setOptions}}
          route={{params: {nextScreen: SCREENS.instruction}}}
        />
      </Provider>,
    );

    await act(async () => {
      fireEvent.press(getByText('Sign In'));
    });

    let modal = getByText('Sign-in failed');
    expect(modal).toBeTruthy();

    fireEvent.press(getByText('Try Again'));

    modal = queryByText('Sign-in failed');
    expect(modal).toBeNull();
  });
});

describe('Instructions screen', () => {
  it('navigates to image capture screen', () => {
    const {getByText} = render(
      <Provider store={store}>
        <Instruction navigation={{navigate, setOptions}} />
      </Provider>,
    );

    fireEvent.press(getByText('Create my face template now'));
    expect(navigate).toHaveBeenCalledWith(SCREENS.imageCapture);
  });
});

describe('Receipt screen', () => {
  it('renders correctly', () => {
    const rendered = create(
      <Receipt navigation={{navigate, setOptions}} />,
    ).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('navigates back to welcome', () => {
    const {getByText} = render(<Receipt navigation={{dispatch, setOptions}} />);

    fireEvent.press(getByText('Back to home'));
    expect(dispatch).toHaveBeenCalledWith(StackActions.popToTop());
  });
});

describe('Manage profile screen', () => {
  it('renders correctly', () => {
    const rendered = create(
      <Provider store={store}>
        <ManageProfile navigation={{navigate, setOptions}} />
      </Provider>,
    ).toJSON();
    expect(rendered).toBeTruthy();
  });
});
