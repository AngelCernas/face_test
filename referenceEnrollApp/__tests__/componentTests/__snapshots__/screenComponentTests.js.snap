// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#002C55",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "#002C55",
          "flex": 1,
          "justifyContent": "center",
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/bg_heroIllustration_welcome.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 0,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderRadius": 4,
            "flex": 1,
            "flexDirection": "column",
            "margin": 16,
            "padding": 15,
          },
          Object {
            "height": "50%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#77787B",
              "fontFamily": "Roboto",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
            }
          }
        >
          Contoso | Real Estate & Security
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
            "marginTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Roboto",
                "fontSize": 24,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 28,
              }
            }
          >
            An easier way to get into work
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#77787B",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 22,
            }
          }
        >
          You can now use face recognition instead of your badge to conveniently unlock building doors.
        </Text>
      </View>
      <View
        style={
          Object {
            "color": "red",
            "paddingBottom": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0078D4",
                  "borderRadius": 4,
                  "paddingHorizontal": 15,
                  "paddingVertical": 15,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 22,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                      },
                      undefined,
                    ]
                  }
                >
                  Get started
                </Text>
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "color": "red",
            "paddingBottom": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#0078D4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "paddingHorizontal": 15,
                  "paddingVertical": 15,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 22,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#0078D4",
                      },
                      undefined,
                    ]
                  }
                >
                  Manage profile
                </Text>
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#77787B",
              "fontFamily": "Roboto",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
            }
          }
        >
          Details at contoso.com/touchless-access
          


          Contoso Privacy Statement
        </Text>
      </View>
    </View>
    <View />
  </View>
</View>
`;
